src/main/java/hw10/FileLoggerConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package hw10;
+
+import java.io.File;
+import java.util.Objects;
+
+public class FileLoggerConfiguration {
+    private final File file;
+    private final LoggingLevel loggingLevel;
+    private final long maxFileSize;
+    private final String messageFormat;
+
+    public FileLoggerConfiguration(File file, LoggingLevel loggingLevel, long maxFileSize, String messageFormat) {
+        Objects.requireNonNull(file);
```

Comment:
You can inline the check and assignments, like this:
`this.file = Objects.requireNonNull(file);`
Do it for other parameters you're checking for null.